api/controllers/document.js

```diff
@@ -671,4 +676,56 @@ exports.protectedDelete = async function (args, res, next) {
     console.log("Error:", e);
     return Actions.sendResponse(res, 400, e);
   }
-};
\ No newline at end of file
+};
+
+exports.featureDocument = async function (args, res, next) {
+  try
+  {
+    if (args.swagger.params.docId && args.swagger.params.docId.value) 
+    {
+      let document = await mongoose.model('Document').findById(mongoose.Types.ObjectId(args.swagger.params.docId.value));
+      let project = await mongoose.model('Project').findById(mongoose.Types.ObjectId(document.project));
+      
+      if(project)
+      {
+        let featuredDocumentsCount = await mongoose.model('Document').count({ project: project._id, isFeatured: true }); 
+
+        // Move the magic number into a config
+        if(featuredDocumentsCount < 5)
+        {
+          document.isFeatured = true;
+          let result = await document.save();
+
+          return Actions.sendResponse(res, 200, result);
+        }
```

Comment:
Do we want to send an explicit error back if they try to favourite a document that pushes it over the limit?

Reply:
RIght now we're not. Validations throw 500 which has a unique error message in the toast and any other failure is "assumed" to be a blocked favorite. I can change it to return a status that we can message on (this is also the plan on the V2 endpoints as well, all endpoint failures return a readable message).